src/firstiteration/Elevator.java

```diff
@@ -0,0 +1,26 @@
+package firstiteration;
+
+import java.util.Date;
+
+public class Elevator implements Runnable{
+	
+	private Scheduler scheduler;
+	
+	public Elevator(Scheduler scheduler) {
+		this.scheduler = scheduler;
+	}
+
+	@Override
+	public void run() {
+		while(true) {
+			//Elevator calling the scheduler
+			RequestData requestData = new RequestData(scheduler.processRequest().getTime(), scheduler.processRequest().getCurrentFloor(),
+					scheduler.processRequest().getDirection(), scheduler.processRequest().getRequestedFloor());
+			//Elevator sends the info back
+			Date date = new Date(System.currentTimeMillis());
+			scheduler.completeRequest(date, requestData.getRequestedFloor(), true);

```

Comment:
my main intention for having this param `currentFloor` passed by elevator was so that elevator can have its current position. And since it'll be moving in real time in whatever motion, it updates its own position (ie. floor number). I think we should make a for loop where it updates its current position attribute till it reaches the request floor (and then keep going till it arrives at destination floor). Reason being is because in next iterations we'll probably have to make some sort of logic for it actually moving. I think you could leave what you have for now but I think it would be cooler to show some progress made in moving to a floor to pickup and then moving to a floor for drop off. And then pass its current position (attribute) to the scheduler.

Reply:
See latest commit for the implementation of this